packages/versioning/src/versioning.ts

```diff
@@ -922,13 +921,34 @@ export function hasDifferentReturnTypeAtVersion(
   return getReturnTypeBeforeVersion(p, type, version) !== "";
 }
 
+function findVersionMapForNamespace(program: Program, ns: Namespace): VersionMap | undefined {
+  const [_, versionMap] = getVersions(program, ns);
+  if (versionMap) {
+    return versionMap;
+  }
+  for (const [_, subNs] of ns.namespaces) {
+    const versionMap = findVersionMapForNamespace(program, subNs);
```

Comment:
why return the versionMap for sub namespaces when asking for the versionMap for a namespace?  Do we have some validation that would prevent multiple sub-namespaces from having different version maps?

Reply:
We have to iterate through the subnamespaces because otherwise it will only look for the enum on the top level namespaces (for example: global, Azure and TypeSpec) when the actual enum is defined deeper within a subnamespace. Because we have no mapping from the enum to the namespaces that depend on it (that was the reverse namespace map in the previous implementation) this is how we need to search for the relevant match.